### ISSUE TITLE
Theme one-dark is not included in this bundle. You may want to load it from external source.

### ISSUE THREAD

**Issue by Serpentarius13** [shikijs/shiki#799]
Hey. I dunno if its a bug or not, but why can I use this theme as defined by types, but it is not being loaded? How to load this theme?
![image](https://github.com/user-attachments/assets/6430ecb9-811c-4095-a1dc-77547e2da380)

**Comment by zhangmo8** [comment]
await highlighter.loadLanguage('xxx')

**Comment by antfu** [comment]
We temporarily closed this due to the lack of enough information. We could not identify whether it was a bug or a userland misconfiguration with the given info.
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue.
Thanks.

> [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by Yatekii** [comment]
This happens to me in a production build but not in a dev build. Not really clear how to provide more info. Production build is here: https://github.com/probe-rs/webpage/pull/165 and the preview: https://c9c6e278.webpage-5t5.pages.dev/. When I build this state locally it all works great :)

**Comment by DerTimonius** [comment]
same thing for me, I am trying to use shiki-magic-move (in a Svelte island for an Astro static site). Dev mode works fine, building and previewing it locally already shows the same error described above. 

This of course also happens when trying to build the prod instance

**Comment by Yatekii** [comment]
You can configure expressive code like so `expressiveCode({ themes: ['vitesse-light', 'vitesse-dark'] })` in the Astro config to fix this :) (of course you have to swap out the themes with the ones you want to use :))

**Comment by dschuessler** [comment]
I ran into the same issue, also with a Svelte island on an Astro static site. For me too, the bug didn't appear when using the local dev server but showed itself on production (Cloudflare Pages).

I managed to both reproduce it and find a workaround. Here is the repository with the minimal reproduction: https://github.com/dschuessler/shiki-bug Clone it and follow the steps in the README.md to see the error.

In my case, I use shiki both with `expressive-code` and `@shikijs/monaco`. The error appeared on all pages where I embedded the monaco editor.

For me, the workaround is to set `expressiveCode({ removeUnusedThemes: false })` in `astro.config.mjs`.
